src/dtx/dtx_resync.c

```diff
@@ -129,6 +128,88 @@ dtx_resync_commit(uuid_t po_uuid, struct ds_cont_child *cont,
 	return rc;
 }
 
+static bool
+dtx_target_alive(struct ds_pool *pool, uint32_t id)
+{
+	struct pool_target	*target;
+	int			 rc;
+
+	rc = pool_map_find_target(pool->sp_map, id, &target);
+	D_ASSERT(rc == 1);
+
+	return target->ta_comp.co_status == PO_COMP_ST_UPIN ? true : false;
+}
```

Comment:
Looks like this is called from target xstreams. Should we take sp_lock when reading sp_map? The system xstream may free sp_map and point it to another pool_map object while target xstreams are reading the previous pool_map object.

There is at least one other sp_map access in dtx_status_handle.

Reply:
You are right. The "sp_map" needs to be protected when access. I will fix related issue (including the one in this patch) in the subsequent patch.